File: SDKs/googleapis.com/dfareporting/v3.3/typescript/src/sdk/models/shared/dimensionvaluelist.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
import { DimensionValue } from "./dimensionvalue";


// DimensionValueList
/** 
 * Represents the list of DimensionValue resources.
**/
export class DimensionValueList extends SpeakeasyBase {
  @Metadata({ data: "json, name=etag" })
  etag?: string;

  @Metadata({ data: "json, name=items", elemType: shared.DimensionValue })
  items?: DimensionValue[];

  @Metadata({ data: "json, name=kind" })
  kind?: string;

  @Metadata({ data: "json, name=nextPageToken" })
  nextPageToken?: string;
}

File: SDKs/googleapis.com/dfareporting/v3.3/go/pkg/models/shared/dimensionvaluelist.go
package shared

// DimensionValueList
// Represents the list of DimensionValue resources.
type DimensionValueList struct {
	Etag          *string          `json:"etag,omitempty"`
	Items         []DimensionValue `json:"items,omitempty"`
	Kind          *string          `json:"kind,omitempty"`
	NextPageToken *string          `json:"nextPageToken,omitempty"`
}
